### PR TITLE
[Static reference] LPS-182671 Remove static references in class MBDiscussionPermission

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:message-boards

**Comment by W2jx** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
no regressions found.

SF is unrelated

**Comment by W2jx** [comment]
no conflicts(7/4 14:30)

**Comment by W2jx** [comment]
no conflicts(7/5 13:57)

**Comment by W2jx** [comment]
no conflicts(7/6 8:08)

File: modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBMessageServiceImpl.java [hunk 1]
```diff
@@ -101,10 +103,12 @@ public MBMessage addDiscussionMessage(
...
-		MBDiscussionPermission.check(
-			getPermissionChecker(), user.getCompanyId(),
-			serviceContext.getScopeGroupId(), className, classPK,
-			ActionKeys.ADD_DISCUSSION);
+		DiscussionPermission discussionPermission =
+			_commentManager.getDiscussionPermission(getPermissionChecker());
+
+		discussionPermission.checkAddPermission(
+			user.getCompanyId(), serviceContext.getScopeGroupId(), className,
+			classPK);
```
Comment (yuhai) [inline modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBMessageServiceImpl.java:108]: @W2jx, the changed logic is different from the original logic. For the original logic, when throw PrincipalException.MustHavePermission(), the userId is from permissionChecker.getUserId(). After the commit, the userId is 0.

File: portal-kernel/src/com/liferay/portal/kernel/comment/DiscussionPermission.java [hunk 2]
```diff
@@ -53,6 +53,10 @@ public default boolean hasPermission(Comment comment, String actionId)
 	public boolean hasPermission(long commentId, String actionId)
 		throws PortalException;
 
+	public boolean hasPermission(
+		String actionId, String className, long classPK, long companyId,
+		long groupId);
+
 	public boolean hasSubscribePermission(
```
Comment (yuhai) [inline portal-kernel/src/com/liferay/portal/kernel/comment/DiscussionPermission.java:76]: @W2jx, if we must add the new method, it seems we should make the param **String actionId** as the last param.

**Comment by yuhai** [comment]
@W2jx, for the commit https://github.com/liferay-core-infra/liferay-portal/pull/4910/commits/0377b145375b7833a05c923e522e80915a6491c0. Actually, PermissionChecker can be get from PermissionThreadLocal.getPermissionChecker(). It seems we needn‘t add the param PermissionChecker for every methods. Based on this, I think we may remove the getDiscussionPermission(PermissionChecker) from CommentManager. That is to say, we may use Reference DiscussionPermission to directly get the service.

File: portal-kernel/src/com/liferay/portal/kernel/comment/DiscussionPermission.java (hunk 2 above)
Comment (W2jx) [reply portal-kernel/src/com/liferay/portal/kernel/comment/DiscussionPermission.java:76]: Hi @yuhai , Don't we need to sort by the first letter of the field name?

**Comment by W2jx** [comment]
see https://github.com/liferay-core-infra/liferay-portal/pull/4944

**Pull request closed without merging**